packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -3208,6 +3221,7 @@ class InputDecoration {
         && other.errorStyle == errorStyle
         && other.errorMaxLines == errorMaxLines
         && other.hasFloatingPlaceholder == hasFloatingPlaceholder
+        && typedOther.alwaysFloatLabel == alwaysFloatLabel
```

Comment:
`typedOther` is causing a failure in the analyzer (see the [analyzer output](https://github.com/flutter/flutter/pull/46115/checks?check_run_id=349782000)).